standalone deployment pipeline for Jenkins (#376)
IBM/dbb: Templates/Deploy_Pipeline/Jenkins/README.md

# Jenkins Deployment Pipeline Template

> This template provides a [Jenkinsfile](Jenkinsfile) to set up a **Continuous Deployment (CD) pipeline** for applications managed in a Git repository.

Reviewer: ```suggestion
This template provides a [Jenkinsfile](Jenkinsfile) to set up a **Deployment pipeline** for applications managed in a Git repository.
```

## Overview and Capabilities

This Jenkins pipeline implements the [Git-based process and branching model for mainframe development](https://ibm.github.io/z-devops-acceleration-program/docs/git-branching-model-for-mainframe-dev) and leverages [Common Backend Scripts](https://github.com/IBM/dbb/blob/main/Templates/Common-Backend-Scripts/README.md) along with **IBM Wazi Deploy** to automate deployments of pre-built application packages from an artifact repository into integration, acceptance, and production environments on z/OS.

The pipeline supports **manual or automated triggers** and implements the following stages:

* **Validate Parameters**  
  - Validates pipeline input parameters passed during manual trigger or via API.  
  - Determines the appropriate environment configuration file based on the selected target environment.  

>   - Ensures required parameters (e.g., `application`, `buildId`, `pipelineType`) are defined.

Reviewer: adjust according to above comments about the parameters

* **Generate Plan**  
  - Uses [`wazideploy-generate.sh`](../../Common-Backend-Scripts/wazideploy-generate.sh) to create the deployment plan in z/OS Unix System Services (USS).  

>   - Supports both **build-based** and **release-based** deployments depending on `pipelineType`.

Reviewer: ```suggestion
  - Supports both the deployment of preliminary **build** and **release** packages.
```

* Access to **IBM Wazi Deploy** scripts and **Common Backend Scripts** installed on the deployment agent.
* Appropriate workspace permissions for reading/writing deployment directories and evidence files.

---

## Installation and Setup

* Place the [Jenkinsfile](Jenkinsfile) in the root directory of your project repository.
* Please review the definitions thoroughly with your Jenkins administrator and adjust pipeline environment variables, workspace paths, and script locations according to your environment.

### Required Parameters

| Parameter           | Required                      | Description                                                               |
| ------------------- | ----------------------------- | ------------------------------------------------------------------------- |
| `application`       | Yes                           | Application name to deploy.                                                |
| `buildId`           | Yes                           | Build pipeline ID corresponding to the artifact.                           |

> | `releaseVersion`    | Only for release pipelines    | Release version (e.g., `rel-2.6.0`).                                       |

Reviewer: Please update based on earlier comments.
